### PR TITLE
Tramstation holiday tiles

### PR DESCRIPTION
**Pull request by lessthnthree**
## About The Pull Request

- Moves the colors list global to be used by any holiday or reason, not just pride
- Adds trimlines as a decal type that can be holiday colored (Tramstation)
- Color generation can now be either horizontal or vertical
- Fixes the name of decals. They were all called "corner"
- Adds missing stack types for tram tiles so they work properly when crowbarred/replaced

## Why It's Good For The Game

Randomly selected trimlines looks terrible, Tram now has some pride too. If you pull up tram tiles they don't turn into generic floor tiles anymore.

image

## Changelog

:cl: LT3
code: Tramstation will now partake in holiday decorations
fix: Tram tiles will remain tram tiles when removed/replaced
spellcheck: Tile decals are now called decals, not corners
/:cl:

### CODE REVIEW

**Comment by Mothblocks** [comment]
i dont understand, what makes this different from the pride week tiles now?

File: code/game/objects/effects/decals/turfdecal/tilecoloring.dm [hunk 1]
```diff
@@ -6,19 +6,39 @@
...
 #define PRIDE_ALPHA 60
 
+GLOBAL_LIST_INIT(rainbow_colors, list(
+		COLOR_PRIDE_PURPLE,
+		COLOR_PRIDE_BLUE,
+		COLOR_PRIDE_GREEN,
+		COLOR_PRIDE_YELLOW,
+		COLOR_PRIDE_ORANGE,
+		COLOR_PRIDE_RED,
+	))
```
Comment (Mothblocks) [inline code/game/objects/effects/decals/turfdecal/tilecoloring.dm:16]: dedent by one

**Comment by Mothblocks** [comment]
ok i see, april fools isnt changed.

> Moves the pride floor colors list global to be used by multiple procs or modified for other themed station days

this confuses me, is this for a future improvement? if so can we wait for it then since its trivial to review a static -> glob change?

File: code/game/objects/effects/decals/turfdecal/tilecoloring.dm [hunk 2]
```diff
@@ -6,19 +6,39 @@
...
+	var/turf/turf = get_turf(atom)
+	return GLOB.rainbow_colors[(turf.y % GLOB.rainbow_colors.len) + 1]
+
+/// Tram needs x-axis aligned tiles
+/obj/effect/turf_decal/tile/neutral/tram/pride_recolor(atom/atom)
+	var/turf/turf = get_turf(atom)
+	return GLOB.rainbow_colors[(turf.x % GLOB.rainbow_colors.len) + 1]
+
+/obj/effect/turf_decal/trimline/tram/pride_recolor(atom/atom)
+	var/turf/turf = get_turf(atom)
```
Comment (Mothblocks) [inline code/game/objects/effects/decals/turfdecal/tilecoloring.dm:40]: You shouldn't need this, every atom has x/y/z vars

**Comment by Mothblocks** [comment]
NVM I am stupid I missed how the code works, carry on

**Comment by lessthnthree** [comment]
> i dont understand, what makes this different from the pride week tiles now?

The normal proc increments colors based on the X value of the turf. On the tram this means it would all be red since the X value is constant. 

> ok i see, april fools isnt changed.
> 
> > Moves the pride floor colors list global to be used by multiple procs or modified for other themed station days
> 
> this confuses me, is this for a future improvement? if so can we wait for it then since its trivial to review a static -> glob change?

Sure. I can change it.

**Comment by lessthnthree** [comment]
> i dont understand, what makes this different from the pride week tiles now?

The proc generates rainbow tiles based on the Y value of the turf. In tram's case the Y value is constant so it ends up being fully red. This swaps it to use the X value instead, generating a rainbow.

File: code/game/objects/effects/decals/turfdecal/tilecoloring.dm (hunk 2 above)
Comment (lessthnthree) [reply code/game/objects/effects/decals/turfdecal/tilecoloring.dm]: Yep, removed.

File: code/game/objects/effects/decals/turfdecal/tilecoloring.dm [hunk 3]
```diff
@@ -7,12 +7,12 @@
...
-		COLOR_PRIDE_YELLOW,
-		COLOR_PRIDE_ORANGE,
-		COLOR_PRIDE_RED,
+	COLOR_PRIDE_PURPLE,
+	COLOR_PRIDE_BLUE,
+	COLOR_PRIDE_GREEN,
+	COLOR_PRIDE_YELLOW,
+	COLOR_PRIDE_ORANGE,
+	COLOR_PRIDE_RED,
 	))
```
Comment (Mothblocks) [inline code/game/objects/effects/decals/turfdecal/tilecoloring.dm:16]: this too

**Comment by optimumtact** [comment]
does this replace the pride effect on all servers?

I don't want two systems to do the same thing

**Comment by lessthnthree** [comment]
> does this replace the pride effect on all servers?
> 
> I don't want two systems to do the same thing

Yes. Most of it is the same code as runs today, the notable change being the /list of colors is turned into a Global so it can be shared with types other than /obj/effect/turf_decal/tile/ (where the list is currently.)

It was in the back of my mind to rename it, since it's not a 'pride rainbow' specific proc, it can be used to color tiles or other objects with whatever flag you want. (For example, red, white, and blue for USA) so I think I'll go ahead and do that.

**Comment by lessthnthree** [comment]
Going to atomize this into smaller components.

**Pull request closed without merging**